FirebaseMessaging/Sources/FIRMessagingAnalytics.m

```diff
@@ -201,9 +202,12 @@ + (void)logMessage:(NSDictionary *)notification
   UIApplicationState applicationState = application.applicationState;
   switch (applicationState) {
     case UIApplicationStateInactive:
-      // App was either in background(suspended) or inactive and user tapped on a display
-      // notification.
-      [self logOpenNotification:notification toAnalytics:analytics];
+      // App was in background and in transition to open when user tapped
+      // on a display notification.
+      // Needs to check notification is displayed.
+      if ([[self class] isDisplayNotification:notification]) {
+        [self logOpenNotification:notification toAnalytics:analytics];
```

Comment:
From #8128, it sounds like we're not logging enough notifications to analytics.  Won't adding the conditional make the log count even less?

Reply:
The reason that we are not logging enough notification open is we are missing the default condition below:
When app is open from shutdown state, this condition is no longer true but the default condition below, which is the background state. (btw, this is weird cuz we haven't changed the code for a long time)


Because of that, only the notifications open when app is in background is logged, not when it's shutdown state. The Application State names here are confusing but the condition did get hit to default state when app is open from shutdown state.

Reply:
Thanks for the explanation!